pkg/kubectl/cmd/auth/reconcile.go

```diff
@@ -87,6 +90,9 @@ func NewCmdReconcile(f cmdutil.Factory, streams genericclioptions.IOStreams) *co
 	o.PrintFlags.AddFlags(cmd)
 
 	cmdutil.AddFilenameOptionFlags(cmd, o.FilenameOptions, "identifying the resource to reconcile.")
+	cmd.Flags().BoolVar(&o.DryRun, "dry-run", false, "If true, display results but do not submit changes")
```

Comment:
> I am wondering if printing needs to account for this somehow.

I don't think that printers have dry run as a first level option.  This the the domain of the command which is controlling behavior. The only printing side-effect is the name printer string which is already configurable.

@juanvallejo and @soltysh since we already litigated this once.

Reply:
Yup, we've agreed that printers don't have to do anything with dry-run, it's command specific.